base/poco/Foundation/include/Poco/AutoPtr.h

```diff
@@ -256,39 +257,51 @@ class AutoPtr
 
     bool operator==(const AutoPtr & ptr) const { return _ptr == ptr._ptr; }
 
-    bool operator==(const C * ptr) const { return _ptr == ptr; }
+    template <typename U>
+    std::enable_if_t<std::is_same_v<std::remove_const_t<U>, C *>,
+    bool> operator==(U ptr) const { return _ptr == ptr; }
```

Comment:
This is a quick fix for creating `AutoPtr<const C>`. Otherwise two same operators are generated from templates. We wanted to use it in ggmlEvaluate.cpp to create a subconfig view like this: https://github.com/ClickHouse/ClickHouse/pull/63668/files#diff-8c7815523803ab1e5abf298a33e6c11c12b8aa400d8b6c398a292710dd991fd8R143. Maybe there is a better way?